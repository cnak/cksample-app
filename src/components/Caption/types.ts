export interface CaptionProps {
    size: 'xl' | 'l' | 'm';
}
