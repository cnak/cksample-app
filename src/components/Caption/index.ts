import styled from 'styled-components/native';

import StyledText from '../StyledText';
import { SECONDARY_TEXT_COLOUR } from '../colors';

import { CaptionProps } from './types';

const fontSizes = {
    xl: 18,
    l: 16,
    m: 16
};

const marginBottom = {
    xl: 5,
    l: 5,
    m: 0
};

const Caption = styled(StyledText)`
    color: ${SECONDARY_TEXT_COLOUR};
    font-size: ${({ size }: CaptionProps) => fontSizes[size]}px;
    margin-bottom: ${({ size }: CaptionProps) => marginBottom[size]}px;
`;

export default Caption;
