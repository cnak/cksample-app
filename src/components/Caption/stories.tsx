import { storiesOf } from '@storybook/react-native';
import React from 'react';

import Caption from '.';

storiesOf('Components', module)
    .add('Caption-XL', () => <Caption size="xl">Creating a new record</Caption>)
    .add('Caption-L', () => <Caption size="l">Creating a new record</Caption>)
    .add('Caption-M', () => <Caption size="m">Creating a new record</Caption>);
