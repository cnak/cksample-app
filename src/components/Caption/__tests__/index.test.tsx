import React from 'react';
import renderer from 'react-test-renderer';

import Caption from '..';

describe('<Caption>', () => {
    describe('matches snapshots', () => {
        it('size "xl"', () => {
            const tree = renderer.create(<Caption size="xl">Label me</Caption>).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('size "l"', () => {
            const tree = renderer.create(<Caption size="l">Label me</Caption>).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('size "m"', () => {
            const tree = renderer.create(<Caption size="m">Label me</Caption>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
