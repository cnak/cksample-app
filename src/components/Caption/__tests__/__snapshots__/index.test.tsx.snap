// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caption> matches snapshots size "l" 1`] = `
<Text
  size="l"
  style={
    Array [
      Object {
        "color": "#6F777B",
        "fontFamily": "Helvetica",
        "fontSize": 16,
        "marginBottom": 5,
      },
    ]
  }
>
  Label me
</Text>
`;

exports[`<Caption> matches snapshots size "m" 1`] = `
<Text
  size="m"
  style={
    Array [
      Object {
        "color": "#6F777B",
        "fontFamily": "Helvetica",
        "fontSize": 16,
        "marginBottom": 0,
      },
    ]
  }
>
  Label me
</Text>
`;

exports[`<Caption> matches snapshots size "xl" 1`] = `
<Text
  size="xl"
  style={
    Array [
      Object {
        "color": "#6F777B",
        "fontFamily": "Helvetica",
        "fontSize": 18,
        "marginBottom": 5,
      },
    ]
  }
>
  Label me
</Text>
`;
